### PR TITLE
Improve error messages when nodes can't communicate with each other.

### PR DESCRIPTION
**Pull request by richardliaw**
Added printing for user to understand what the program is doing when nodes can't communicate.

### CODE REVIEW

File: python/ray/worker.py [hunk 1]
```diff
@@ -674,7 +674,9 @@ def get_address_info_from_redis(redis_address, node_ip_address, num_retries=5):
       if counter == num_retries:
         raise
       # Some of the information may not be in Redis yet, so wait a little bit.
-      print("Some processes that the driver needs to connect to have not registered with Redis, so retrying.")
+      print("Some processes that the driver needs to connect to have not "
+            "registered with Redis, so retrying. Have you run "
```
Comment (richardliaw) [inline python/ray/worker.py]: should this be placed outside the for loop?

Comment (richardliaw) [inline python/ray/worker.py]: specifically the last part "have you tried..."

**Review by richardliaw — COMMENTED** [review]
nit

File: python/ray/worker.py (hunk 1 above)
Comment (robertnishihara) [inline python/ray/worker.py]: I addressed this comment, but now I'm concerned that the user won't see the message telling them to run `scripts/start_ray.sh`... seeing the message multiple times is annoying, but this way might be worse.

**Pull request merged**